pkg/istio-agent/agent.go

```diff
@@ -546,10 +546,14 @@ func (a *Agent) GetDNSTable() *dnsProto.NameTable {
 				if _, exists := nt.Table[host]; !exists {
 					addresses := make([]string, len(ipv4)+len(ipv6))
 					for _, addr := range ipv4 {
-						addresses = append(addresses, addr.String())
+						if len(addr.String()) > 0 {
```

Comment:
`ipv4` slice item here should not be empty

Reply:
@zhlsunshine

Reply:
```
        "a-benjamin-sandbox-test-ingress.armada-a-benjamin-sandbox-test.": {
            "ips": [
                "",
                "172.20.216.52"
            ],
            "registry": "Kubernetes"
        },
        "a-benjamin-sandbox-test-ingress.armada-a-benjamin-sandbox-test.svc.": {
            "ips": [
                "",
                "172.20.216.52"
            ],
            "registry": "Kubernetes"
        },
```
It happens for alt hosts in. Let me debug this a bit more

Reply:
https://github.com/istio/istio/blob/1607f91e1c5c85b6927bebca591b9aafc4b16e05/pkg/dns/client/dns.go#L600 - We handle this case while building actual DNS responses. But here it is missed.

Reply:
Ok. I debugged this more, it is not with empty IPs but with len vs. capacity :-)

Reply:
> Ok. I debugged this more, it is not with empty IPs but with len vs. capacity :-)

Hi @hzxuzhonghu & @hzxuzhonghu, does it mean that there may be some invalidation IP entries?

Reply:
It is only in debug endpoint, not on the actual code